### ISSUE TITLE
nil pointer dereference on $ref without # character

### ISSUE THREAD

**Issue by xartishuk** [daveshanley/vacuum#225]
### Issue
When running `vacuum lint openapi.yaml` I get a "nil pointer dereference" error. 
In the `openapi.yaml` I have the following `$ref`: `$ref: 'common/parameters/authorization.yaml'`. If the `$ref` contains a `#` character at the end of the file name, e.g. `$ref: 'common/parameters/authorization.yaml#'` - the issue is gone and `lint` produces normal output.

#### Full error output:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xaec5f2]

goroutine 81 [running]:
github.com/daveshanley/vacuum/functions/openapi.OperationParameters.RunRule({}, {0xc000532090?, 0xf0e080?, 0x1?}, {0xc0000ad410, 0xc0001048f0, {0xf0a600, 0x11ee520}, {0x0, 0x0}, ...})
	/home/some-path/vacuum/functions/openapi/operation_parameters.go:73 +0xd72
github.com/daveshanley/vacuum/motor.buildResults({0xc0001048f0, 0xc0003c2dc0, {0x11f39b0, 0xc0003f4a60}, 0xc0003f60f0, 0xc0003ded10, 0xc0003f6060, 0xc000580400, 0xc0003c2e60, 0x0, ...}, ...)
	/home/some-path/vacuum/motor/rule_applicator.go:405 +0x61c
github.com/daveshanley/vacuum/motor.runRule({0xc0001048f0, 0xc0003c2dc0, {0x11f39b0, 0xc0003f4a60}, 0xc0003f60f0, 0xc0003ded10, 0xc0003f6060, 0xc000580400, 0xc0003c2e60, 0x0, ...})
	/home/some-path/vacuum/motor/rule_applicator.go:330 +0x40f
created by github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet
	/home/some-path/vacuum/motor/rule_applicator.go:165 +0x72e
```
#### Full contents of `openapi.yaml`:
```yaml
---
openapi: 3.0.3
info:
  title: My service
  description: Nil pointer dereference service.
  version: 1.0.0
  contact:
    name: Me
    email: xartishuk@gmail.com

tags:
  - name: Devices
    description: Device operations

paths:
  /v1/devices:
    post:
      tags:
        - Devices
      summary: Create a device
      description: Create a device resource.
      operationId: devices::create
      parameters:
        - $ref: 'common/parameters/authorization.yaml'
      responses:
        '200':
          description: OK
          content:
            application/vnd.api+json:
              schema:
                type: object
                properties:
                  meta:
                    type: object
                    properties:
                      message:
                        type: string
                        example: 'OK'
```
#### Full contents of `common/parameters/authorization.yaml`:
```yaml
name: Authorization
in: header
required: true
schema:
  type: string
  example: Bearer <JWT token>
```

**Comment by daveshanley** [comment]
Looking into this for you.

**Comment by daveshanley** [comment]
This should be resolved in `v0.0.49`

**Comment by xartishuk** [comment]
Awesome, thanks for looking into it
